File: python/162. Find Peak Element.py
class Solution(object):
    def findPeakElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        l = 0
        r = n-1
        while l<r:
            m1 = l + (r-l)/2
            m2 = m1+1
            if nums[m1] < nums[m2]:
                l = m2
            else:
                r = m1
        return l

File: go/src/leetcode/162. Find Peak Element/162. Find Peak Element.go
package main

import "fmt"

func findPeakElement(nums []int) int {
	n := len(nums)
	l, r := 0, n-1
	var m int
	for l < r {
		m = l + (r-l)>>1
		if nums[m+1] < nums[m] { // 6,5
			r = m
		} else {
			l = m + 1
		}
	}
	return l
}

func main() {
	fmt.Println(findPeakElement([]int{1,2,3,1}))
	fmt.Println(findPeakElement([]int{1,2,1,3,5,6,4}))
}
